net/bonding: stop all slaves when stopping bonded port

When stopping a bonding device, stop all slaves. There
may be slaves that are started and link status not up
yet and slaves not activated when the bonded port is
being stopped.

Fixes: 0911d4ec01 (net/bonding: fix crash when stopping mode 4 port"

Signed-off-by: Sharmila Podury <sharmila.podury@att.com>
--- a/drivers/net/bonding/rte_eth_bond_pmd.c
+++ b/drivers/net/bonding/rte_eth_bond_pmd.c
@@ -2104,11 +2104,11 @@ bond_ethdev_stop(struct rte_eth_dev *eth
 	internals->link_status_polling_enabled = 0;
 	for (i = 0; i < internals->slave_count; i++) {
 		uint16_t slave_id = internals->slaves[i].port_id;
+		internals->slaves[i].last_link_status = 0;
+		rte_eth_dev_stop(slave_id);
 		if (find_slave_by_id(internals->active_slaves,
 				internals->active_slave_count, slave_id) !=
 						internals->active_slave_count) {
-			internals->slaves[i].last_link_status = 0;
-			rte_eth_dev_stop(slave_id);
 			deactivate_slave(eth_dev, slave_id);
 		}
 	}
